src/ethereum/EthereumVerifier.js

```diff
@@ -337,8 +338,8 @@ class EthereumVerifier {
   _verifySelectedWorker(secretContractAddress, workerAddress, params) {
     let result = {error: null, isVerified: true};
     let selectedWorker = EthereumVerifier.selectWorkerGroup(secretContractAddress, params, 1)[0];
-    // selectedWorker = nodeUtils.remove0x(selectedWorker.signer.toLowerCase());  // To enable in update to the contract
-    if (selectedWorker.signer !== workerAddress) {
+    selectedWorker = nodeUtils.remove0x(selectedWorker.toLowerCase());
```

Comment:
Could you please elaborate why the call to `toLowerCase()` ?

Reply:
Because `core` handles and outputs all its addresses in lowercase whereas the contract returns them as checksumAddress. In order to match `core`, the client also lowercases all addresses contained in any request so `workerAddress` will come in lowercase, whereas the address that is stored in workerParams is a checksumAddress, and there wouldn't be a match otherwise due to the mismatched lowercase vs lower+uppercase.